arch/x86/core/x86_mmu.c

```diff
@@ -1774,8 +1774,8 @@ static inline int apply_region(pentry_t *ptables, void *start,
 __pinned_func
 static void set_stack_perms(struct k_thread *thread, pentry_t *ptables)
 {
-	LOG_DBG("update stack for thread %p's ptables at %p: %p (size %zu)",
-		thread, ptables, (void *)thread->stack_info.start,
+	LOG_DBG("update stack for thread %p's ptables at %p: 0x%" PRIxPTR " (size %zu)",
+		thread, ptables, thread->stack_info.start,
 		thread->stack_info.size);
 	apply_region(ptables, (void *)thread->stack_info.start,
```

Comment:
Seems like this is just covering the case in the log function, but not the one on line 1780. I guess it'd be better for the code to be consistent, i.e. avoid the cast everywhere or let it be (after all, `uintptr_t` should always be safe to cast to a pointer type).

Reply:
If I do that the tests fail